cmd/wallet-web/src/pages/mixins/common/helper.js

```diff
@@ -89,6 +89,50 @@ export const isVPType = (type) => toLower(type) == 'verifiablepresentation';
 export const getCredentialType = (types) =>
   types.filter((type) => type != 'VerifiableCredential')[0];
 
+// function to replace json path from the json objects
+export function getCredentialDisplayData(vc, manifest) {
+  // title
+  const titleFound = findByPath(vc, manifest.title.path);
+  const title = titleFound ? titleFound : manifest.title.fallback;
+
+  // icon
+  const icon = manifest.icon;
+
+  // find properties
+  const _readProperty = (property) => {
+    const pKey = property[0];
+    const vals = findByPath(vc, property[1].path);
+    const value = vals ? vals : '';
+    const label = property.label;
+    const type = property.type;
+    const format = property.format;
+    return {
+      [pKey]: {
+        label,
+        value,
+        type,
+        format,
+      },
+    };
+  };
+
+  const properties = Object.entries(manifest.properties).map(_readProperty);
+  return {
+    title,
+    icon,
+    properties,
+  };
+}
+// function to replace json path from the json objects
+function findByPath(vc, ...paths) {
+  let pathValue;
+  for (const key in paths) {
+    pathValue = jp.query(vc, paths[key][0]);
+  }
+  // returns the first value which is path[0].
+  return pathValue.shift();
```

Comment:
Just return `pathValue[0]` - you don't need to modify `pathValue` array

Reply:
I'm still not clear about this function. We should only call `jp.query` on `path` properties, not on every path. Besides, you only pass single string to `paths` on lines https://github.com/trustbloc/wallet/pull/1005/files#diff-901fbc10d64331f455b4ac2aebc1f882cb8a5fded27e30dadc624a917e75ec1cR104 and https://github.com/trustbloc/wallet/pull/1005/files#diff-901fbc10d64331f455b4ac2aebc1f882cb8a5fded27e30dadc624a917e75ec1cR95 - so what's the type of `paths`? Should we rename it to `path` if it's just a single string?